Refactor coming soon templates to use dynamic font families

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #54049

This PR updates the "Coming Soon" templates to dynamically apply the font family based on the active theme. It ensures that the fonts for headings and paragraphs adapt to the current block theme’s typography while maintaining specific behavior for non-block themes and the TwentyTwentyFour theme.


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with `coming-soon-newsletter-template` feature flag enabled.
2. Enable coming soon mode in site visibility settings.
3. Test different themes:

-   TwentyTwentyFour Theme:
    -   Activate the TwentyTwentyFour theme.
    -   Go to `Appearance > Editor > Templates > Coming soon`
    -   Select any template from the sidebar panel  (Template > Design)
    -   Verify that headings and paragraphs in "Coming Soon" templates use Cardo or Inter or SF_Pro (--apple-system) fonts.

![Screenshot 2025-01-15 at 16 57 20](https://github.com/user-attachments/assets/090147ae-fa2b-463a-bfa9-f0def6880737)


-   Block Themes:
    -   Activate a block theme (other than TwentyTwentyFour)
    -   Navigate to the "Coming Soon" template editor.
    -   Verify that headings and paragraphs use the theme's defined fonts.


When using Next Bakery Theme:

![Screenshot 2025-01-15 at 16 56 32](https://github.com/user-attachments/assets/d3afc7d5-ecc1-4edb-bf63-743676f70495) 


-   Non-Block Themes:
    -   Go to `Appearance > Editor > Templates` and reset the coming soon template.
    -   Activate a non-block theme.
    -   Go to the site's front page in an incognito window and verify that headings and paragraphs use Cardo or Inter fonts, consistent with the TwentyTwentyFour theme styling.

`Template part has been deleted or is unavailable: coming-soon-social-links` is a bug that we should fix in another PR.

![Screenshot 2025-01-15 at 17 01 17](https://github.com/user-attachments/assets/09e71c76-1a9b-482b-8e02-73e33ab53c94)



<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Updated "Coming Soon" templates to dynamically adapt font families based on the active theme, with specific behavior for TwentyTwentyFour and non-block themes

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan committed:
Refactor coming soon templates to utilize dynamic font families

- Updated multiple coming soon templates to use the new ComingSoonTemplate class for font family retrieval.
- Removed hardcoded font family assignments and replaced them with dynamic values based on the current theme.
- Enhanced the styling of site titles, headings, and paragraphs to reflect the selected font families.
- Improved maintainability and consistency across coming soon templates by centralizing font management.

This change ensures that the templates adapt better to different themes and provide a more cohesive design experience.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

chihsuan committed:
Refactor coming soon templates to enhance font management

- Updated the coming soon templates to utilize dynamic font families retrieved from the new ComingSoonTemplate class.
- Removed hardcoded font family assignments, allowing for better adaptability to different themes.
- Improved the styling of headings and paragraphs to reflect the selected font families, ensuring a cohesive design experience.
- Enhanced maintainability by centralizing font management, making future updates easier.

This change builds on previous efforts to improve template flexibility and design consistency.

psealock reviewed (approved):
Code looks good and I can see the fonts updated when I change themes. Nice one @chihsuan 🚢

ilyasfoo commented on plugins/woocommerce/src/Blocks/Templates/ComingSoonTemplate.php:
```diff
@@ -50,4 +50,58 @@ protected function get_placeholder_page() {
+			);
+		}
+
+		if ( ! function_exists( 'wp_get_global_settings' ) ) {
+			require_once ABSPATH . WPINC . '/global-styles-and-settings.php';
+		}
```
I'm hesitant about trying to load a core file manually. If the function is not available, I suspect it must be because the function isn't called within the correct WordPress lifecycle/context. Additionally, core paths may change in the future, making this a tech debt.

Since we're just trying to get a font family, I think it's not critical enough to risk a crash or add tech debt. What do you think of just returning `$default_fonts` when the function isn't available?

chihsuan replied:
I think core paths are stable and rarely change, and even if they change, we should be able to update the code accordingly, but your point is valid. Since it's not a critical function, returning $default_fonts is a better solution. I'll make the change. Thanks for the suggestion!

chihsuan replied:
Updated in 3f8f3a37b2b76ff50ad5ba89ba0e59f19bcf3eec

ilyasfoo reviewed (commented):
Thanks for working on this, @chihsuan! I'm leaving a partial review today

> Template part has been deleted or is unavailable: coming-soon-social-links is a bug that we should fix in another PR.

I wasn't able to reproduce this using Storefront, did you use some other theme?

<img width="1251" alt="image" src="https://github.com/user-attachments/assets/e98a6ff5-f6b7-4caf-868d-7d9ca41ffb7d" />

chihsuan commented:
Thanks for the review! @ilyasfoo 

> I wasn't able to reproduce this using Storefront, did you use some other theme?

I've fixed it in https://github.com/woocommerce/woocommerce/pull/54624. 😄

chihsuan committed:
Fix font retrieval in ComingSoonTemplate when global styles are unavailable

- Modify the ComingSoonTemplate to handle cases where WordPress global styles function is not available
- Return default fonts when wp_get_global_settings() cannot be used
- Improve robustness of font management in coming soon templates

ilyasfoo reviewed (approved):
Thanks, @chihsuan! LGTM 🚢

chihsuan committed:
Merge branch 'trunk' into update/coming-soon-template-dynamic-font
